framework/web/Request.php

```diff
@@ -283,21 +311,23 @@ public function getHeaders()
             $this->_headers = new HeaderCollection();
             if (function_exists('getallheaders')) {
                 $headers = getallheaders();
+                foreach ($headers as $name => $value) {
+                    $this->_headers->add($name, $value);
+                }
             } elseif (function_exists('http_get_request_headers')) {
```

Comment:
Again not something introduced by you, but I found this in the source code for the PECL HTTP extension (where `http_get_request_headers()` comes from:

https://github.com/m6w6/ext-http/blob/45fb8dff8bc2981151fbca023408c61acbba0f08/src/php_http_env.c#L36

To me this suggests that anywhere this function is available `$_SERVER` must be available as well. Therefor I think this workaround could be removed.